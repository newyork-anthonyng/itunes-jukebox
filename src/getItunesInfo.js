import axios from "axios";

const constructAlbumUrl = artistName => {
    // https://developer.apple.com/library/content/documentation/AudioVideo/Conceptual/iTuneSearchAPI/Searching.html#//apple_ref/doc/uid/TP40017632-CH5-SW2
    return `https://itunes.apple.com/search?term=${artistName.split(" ").join("+")}&entity=album`;
}

const getAlbumsForArtist = artistName => {
    return axios.get(constructAlbumUrl(artistName));
};

const formatAlbumsInfo = axiosResponse => {
    return {
        albums: axiosResponse.data.results.map(formatAlbumInfo)
    };
};

const formatAlbumInfo = ({ collectionId, collectionName, artistName, artworkUrl100, collectionViewUrl }) => ({
    id: collectionId,
    artistName,
    albumArt: artworkUrl100,
    albumName: collectionName,
    itunesLink: collectionViewUrl
});

export {
    getAlbumsForArtist,
    formatAlbumsInfo
};