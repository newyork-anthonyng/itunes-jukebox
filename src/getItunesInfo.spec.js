import { getAlbumsForArtist, formatAlbumsInfo } from "./getItunesInfo";
jest.mock("axios", () => {
  return {
    get: jest.fn(() => Promise.resolve())
  };
});
import axios from "axios";

describe("getAlbumsForArtist", () => {
  it("should call correct API", () => {
    const result = getAlbumsForArtist("Michael Jackson");

    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get.mock.calls[0][0]).toEqual(
      "https://itunes.apple.com/search?term=Michael+Jackson&entity=album"
    );
    // expect getAlbumsForArtist to return a Promise
    expect(typeof result.then).toEqual("function");
  });
});

describe("formatAlbumsInfo", () => {
  it("should format axios response from iTunes correctly", () => {
    const mockAxiosResponse = {
      data: {
        resultCount: 34,
        results: [
          {
            wrapperType: "collection",
            collectionType: "Album",
            artistId: 32940,
            collectionId: 159292399,
            amgArtistId: 4576,
            artistName: "Michael Jackson",
            collectionName: "The Essential Michael Jackson",
            collectionCensoredName: "The Essential Michael Jackson",
            artistViewUrl:
              "https://itunes.apple.com/us/artist/michael-jackson/32940?uo=4",
            collectionViewUrl:
              "https://itunes.apple.com/us/album/the-essential-michael-jackson/159292399?uo=4",
            artworkUrl60:
              "https://is2-ssl.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/60x60bb.jpg",
            artworkUrl100:
              "https://is2-ssl.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/100x100bb.jpg",
            collectionPrice: 16.99,
            collectionExplicitness: "notExplicit",
            trackCount: 38,
            copyright:
              "â„— 1972 Motown Records, a Division of UMG Recordings, Inc., 1976, 1978, 1980 Sony Music Entertainment, 1979, 1982, 1987, 1991, 1995, 2001, 2005 MJJ Productions Inc.",
            country: "USA",
            currency: "USD",
            releaseDate: "2005-07-05T07:00:00Z",
            primaryGenreName: "Pop"
          }
        ]
      }
    };
    expect(formatAlbumsInfo(mockAxiosResponse)).toMatchSnapshot();
  });
});
