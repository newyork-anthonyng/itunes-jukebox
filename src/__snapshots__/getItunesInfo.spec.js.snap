// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatAlbumsInfo should format axios response from iTunes correctly 1`] = `
Object {
  "albums": Array [
    Object {
      "albumArt": "https://is2-ssl.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/100x100bb.jpg",
      "albumName": "The Essential Michael Jackson",
      "artistName": "Michael Jackson",
      "id": 159292399,
      "itunesLink": "https://itunes.apple.com/us/album/the-essential-michael-jackson/159292399?uo=4",
    },
  ],
}
`;
